src/frontend/src/utils/findWebAuthnRpId.ts

```diff
@@ -0,0 +1,107 @@
+import { DeviceData } from "$generated/internet_identity_types";
+
+const defaultDomain = "https://identity.ic0.app";
+
+/**
+ * Helper to extract the top and secondary level domain from a URL.
+ *
+ * Example: "https://identity.ic0.app" -> "ic0.app"
+ *
+ * @param url {string} The URL to extract the domain from.
+ * @returns {string} The top and secondary level domain.
+ *
+ * @throws {Error} If the URL is invalid.
+ * @throws {Error} If the URL does not contain a top and secondary level domain.
+ */
+const getTopAndSecondaryLevelDomain = (url: string): string => {
+  const parts = new URL(url).hostname.split(".").reverse();
+
+  if (parts.length < 2) {
+    throw new Error("Invalid URL: Unable to extract domain.");
+  }
+
+  return `${parts[1]}.${parts[0]}`;
+};
```

Comment:
The `reverse()` call just to reverse it manually back again in `${parts[1]}.${parts[0]}` is slightly confusing, I would recommend to simply slice the last two parts and join them back together.

```suggestion
const getTopAndSecondaryLevelDomain = (url: string): string => {
  const parts = new URL(url).hostname.split(".");

  if (parts.length < 2) {
    throw new Error("Invalid URL: Unable to extract domain.");
  }

  return parts.slice(-2).join('.');
};
```